sycl/test/devicelib/assert.cpp

```diff
@@ -0,0 +1,200 @@
+// REQUIRES: cpu,linux
+// RUN: %clangxx -fsycl -c %s -o %t.o
+// RUN: %clangxx -fsycl %t.o %llvm_build_libs_dir/libsycl-glibc.o -o %t.out
+// (see the other RUN lines below; it is a bit complicated)
+//
+// assert() call in device code guarantees nothing: on some devices it behaves
+// in the usual way and terminates the program. On other devices it can print an
+// error message and *continue* execution. Less capable devices can even ignore
+// an assert!
```

Comment:
So the conclusion is...

Reply:
> So the conclusion is...

... the assert's behavior in SYCL device code is unspecified. However, you should get the expected behavior on devices that have capabilities to do that.

Reply:
OK. But how to do this in a consistent and reasonable way?

Reply:
You have implement cl_intel_devicelib_cassert extension for all devices you care about to guarantee consistency between them.